tests/suite/upgrade_test.go

```diff
@@ -208,11 +208,12 @@ var _ = Describe("Upgrade testing", Label("nfr", "upgrade"), func() {
 			500*time.Millisecond,
 			true, /* poll immediately */
 			func(_ context.Context) (bool, error) {
+				defer GinkgoRecover()
 				Expect(k8sClient.Get(leaseCtx, key, &lease)).To(Succeed())
 
 				if lease.Spec.HolderIdentity != nil {
 					for _, podName := range podNames {
-						if podName == *lease.Spec.HolderIdentity {
+						if podName == strings.Split(*lease.Spec.HolderIdentity, "_")[0] {
```

Comment:
Thoughts on changing it to something like

```suggestion
						if strings.Contains(*lease.Spec.HolderIdentity, podName) {
```

I think that might be a bit better where we wouldn't need to worry about a stray `_` ending up in the podname or lease holder.